internal/resources/network_peer.go

```diff
@@ -104,6 +105,27 @@ func (n *NetworkPeer) Create(ctx context.Context, req resource.CreateRequest, re
 		}
 		networkPeerRequest.ProviderConfig = providerConfigJSON
 		plan.ProviderConfig.AWSConfig = nil
+		plan.ProviderConfig.AzureConfig = nil
+
+	} else if plan.ProviderConfig.AzureConfig != nil {
+		azureConfigJSON := network_peer_api.AzureConfigData{
+			AzureTenantId:  plan.ProviderConfig.AzureConfig.AzureTenantId.ValueString(),
+			ResourceGroup:  plan.ProviderConfig.AzureConfig.ResourceGroup.ValueString(),
+			SubscriptionId: plan.ProviderConfig.AzureConfig.SubscriptionId.ValueString(),
+			VnetId:         plan.ProviderConfig.AzureConfig.VnetId.ValueString(),
+			Cidr:           plan.ProviderConfig.AzureConfig.Cidr.ValueString(),
+		}
+		providerConfigJSON, err := json.Marshal(azureConfigJSON)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error creating network peer for Azure",
+				errors.ErrConvertingProviderConfig.Error(),
+			)
+			return
+		}
+		networkPeerRequest.ProviderConfig = providerConfigJSON
+		plan.ProviderConfig.AWSConfig = nil
+		plan.ProviderConfig.GCPConfig = nil
 	}
```

Comment:
should handle edge case where aws/gcp/azure configs are nil

Reply:
that case is not possible though, the user can't have none of the configs in the request for network peer.

Reply:
> the user can't have none of the configs (aws/gcp/azure) in the request

It is possible.  I just tested it.  With this config:

```
resource "couchbase-capella_network_peer" "new_network_peer" {
  organization_id = var.organization_id
  project_id      = var.project_id
  cluster_id      = var.cluster_id
  name            = var.network_peer.name
  provider_type   = var.network_peer.provider_type
  provider_config = {}
}
```

Then run `terraform apply`:

```
terraform apply
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - couchbasecloud/couchbase-capella in /Users/$USER/GolandProjects/terraform-provider-couchbase-capella/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # couchbase-capella_network_peer.new_network_peer will be created
  + resource "couchbase-capella_network_peer" "new_network_peer" {
      + audit           = (known after apply)
      + cluster_id      = "60c2e7fb-aa09-4d4f-b53d-7d9965f06fde"
      + commands        = (known after apply)
      + id              = (known after apply)
      + name            = "VNETPeerTFTestAZURE"
      + organization_id = "6af08c0a-8cab-4c1c-b257-b521575c16d0"
      + project_id      = "c1fade1a-9f27-4a3c-af73-d1b2301890e3"
      + provider_config = {}
      + provider_type   = "azure"
      + status          = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value:
```

When I type `yes` I get this error:

```
Error: Error creating network peer
│
│   with couchbase-capella_network_peer.new_network_peer,
│   on config.tf line 105, in resource "couchbase-capella_network_peer" "new_network_peer":
│  105: resource "couchbase-capella_network_peer" "new_network_peer" {
│
│ There is an error during network peer creation. Please check in Capella to see if any hanging resources have been created, unexpected error: {"hint":"Please review your request and ensure that all required
│ parameters are correctly provided.","message":"Unable to create new VPC peer. The required information for the Azure VPC to peer with was not provided.","code":422,"httpStatusCode":422}
```

> won't make sense

You're right it won't make sense.  Hence we should return an error.

Reply:
But we are already throwing this error- `"Unable to create new VPC peer. The required information for the Azure VPC to peer with was not provided.","code":422,"httpStatusCode":422"`

Reply:
Sorry I shouldn't have been more clear.

I mean throw a validation error saying `provider_config` cannot be empty.  No point in sending the request to `open-api` with incomplete info.

Reply:
Okay added an else to handle this, throwing the following error on `terraform apply`-

<img width="1566" alt="Screen Shot 2024-08-28 at 5 36 13 PM" src="https://github.com/user-attachments/assets/2c5977d3-9c2f-4dc7-b4b8-2a2dc5ec710c">